### PR TITLE
add vax location detail drawer

### PR DESCRIPTION
**Pull request by syauqy**
vax detail drawer

I add a drawer to view the vax location detail and vax schedule easier, esp. on the mobile view.
Maybe with this addition, we can clean up some information on the front page.

### CODE REVIEW

File: src/modules/vax/VaxLocationDetail.tsx [hunk 1]
```diff
@@ -153,6 +174,106 @@ export default function VaxLocationDetail({ loading, isUserLocationExist, locati
...
+                        );
+                      })}
+                    </Tbody>
+                  </Table>
+                </Box>
+              ))}
+            </Stack>
+          </DrawerBody>
+        </DrawerContent>
+      </Drawer>
```
Comment (resir014) [inline src/modules/vax/VaxLocationDetail.tsx]: Can we split this drawer into a separate component? `modules/vax/VaxLocationDetailDrawer.tsx`

**Review by resir014 — CHANGES_REQUESTED** [review]
Thank you for this PR @syauqy! This is the first pass of the review. I'll do a further design pass after this review is complete and the build failures + CI errors are fixed.

File: src/modules/vax/VaxLocationDetail.tsx (hunk 1 above)
Comment (syauqy) [reply src/modules/vax/VaxLocationDetail.tsx]: Sure, I'll do it

Comment (syauqy) [reply src/modules/vax/VaxLocationDetail.tsx]: VaxLocationDetailDrawer.tsx added. kindly check the new commit https://github.com/tibudiyanto/jakarta-vax-availability/pull/79/commits/79a3dee10472a35c780a4d761fec646ae7fa33f9

File: src/modules/vax/VaxLocationDetail.tsx [hunk 2]
```diff
@@ -77,10 +80,16 @@ export default function VaxLocationDetail({ loading, isUserLocationExist, locati
...
   };
-
   return (
     <>
-      <Flex alignItems="center" flexDirection="row" justifyContent="space-between" p={4}>
+      <Flex
+        alignItems="center"
+        flexDirection="row"
+        justifyContent="space-between"
+        onClick={onOpen}
```
Comment (resir014) [inline src/modules/vax/VaxLocationDetail.tsx]: **Never** use `onClick` on non-interactable elements (e.g. `div`). `chakra-ui` has a [LinkOverlay component](https://chakra-ui.com/docs/navigation/link-overlay) for a more accessible alternative.

**Review by resir014 — CHANGES_REQUESTED** [review]
@syauqy **Design pass results:**

1. Instead of making the entire card clickable, let's use the space below and add a button there to open the drawer, like this:

image

You can duplicate this block in `VaxLocation` and change the button label to "Lihat Jadwal". Use `CalendarIcon` for the icon, and replace `Link` with an unstyled button (`<Box as="button" />`).

image

2. The space on this header is too tight. Make sure to use `Stack` to evenly space them. Use `space={2}` for the spacing.

image

3. Make sure the text colours have proper contrast in dark mode. Currently the shade of green, and the grey, has too low contrast, so it might be hard to see for people with visual impairments.

image

**Comment by syauqy** [comment]
Thank you for the review @resir014. 
1. May I move `Lihat Lokasi` flex block on `VaxLocation` to `VaxLocationDetail`? 
And since the schedules are inside the drawer, can we remove the schedule button and popover at `VaxLocation` card?

image

2. Header space and text colors on the drawer is fixed. What do you think?

image

I'm not really sure about the green color, is it too bright? Also, what do you think of the table size? is it too tight and small?
For the comparison, this is "md" table size.

image

**Comment by resir014** [comment]
> And since the schedules are inside the drawer, can we remove the schedule button and popover at VaxLocation card?

@syauqy These are still needed in the map page. Either keep it, or add a prop to `VaxLocationDetail` to toggle its visibility on e.g. the map page.

Also, the drawer should be moved to `VaxLocation` since it's not needed in the map page. Everything inside `VaxLocationDetail` is also rendered on the map page.

> I'm not really sure about the green color, is it too bright?

These are fine. Remember that you can also use `useColorModeValue(lightColor, darkColor)` hook to use different colours on light/dark mode.

> Also, what do you think of the table size? is it too tight and small?

Let's keep it as it is.

**Comment by syauqy** [comment]
@resir014 
sure, the drawer moved to `VaxLocation` and I keep the popover.

> Remember that you can also use useColorModeValue(lightColor, darkColor) hook to use different colours on light/dark mode.

Ahh I see, this is helpful. thanks.

I pushed a new commit at https://github.com/tibudiyanto/jakarta-vax-availability/pull/79/commits/4a51660f6dd75333a5b326a6e500973531796c85

File: src/modules/vax/VaxLocationDetailDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            onClose();
+          }}
+          p={4}
+          style={{
+            cursor: 'pointer'
+          }}
+        >
+          <ArrowBackIcon h={4} w={4} />
+          Kembali
+        </Badge>
```
Comment (resir014) [inline src/modules/vax/VaxLocationDetailDrawer.tsx]: This should be wrapped in a `Box` and replaced with a `Button` or `IconButton`. As mentioned earlier, **never** use `onClick` on non-interactable elements (e.g. `div`, `span`).

**Review by resir014 — CHANGES_REQUESTED** [review]
@syauqy Minor changes, mostly accessibility issues:

File: src/modules/vax/VaxLocationDetailDrawer.tsx (hunk 3 above)
Comment (syauqy) [reply src/modules/vax/VaxLocationDetailDrawer.tsx]: ahh, Sorry for that. 
Changed it to `IconButton`. Hope it's okay.

image

d358d555551be3ae8ef741c04f865718b4b73005

**Review by resir014 — APPROVED** [review]
All good now. Thanks!

**Pull request merged**